test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{match: MATCH, method: METHOD}]),
+        'incorrect-type',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{match: MATCH, method: METHOD, handler: {}}]),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{match: MATCH, handle: HANDLER}]),
+        'incorrect-type',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+    });
+
+    it(`should modify the internal arrays of routes when register/unregister is called`, function() {
+      const router = new Router();
+
+      // Routes without an explicit method will default to GET.
+      const getRoute1 = new Route(MATCH, HANDLER);
+      const getRoute2 = new Route(MATCH, HANDLER, 'GET');
+      const putRoute1 = new Route(MATCH, HANDLER, 'PUT');
+      const putRoute2 = new Route(MATCH, HANDLER, 'PUT');
+      // We support passing in Objects that match the expected interface in addition to Routes.
+      const postRoute = {
+        match: MATCH,
+        handler: {handle: HANDLER},
+        method: 'POST',
+      };
+
+      router.registerRoute(getRoute1);
+      router.registerRoutes([getRoute2, putRoute1, putRoute2, postRoute]);
+
+      expect(router._routes.get('GET')).to.have.members([getRoute1, getRoute2]);
+      expect(router._routes.get('PUT')).to.have.members([putRoute1, putRoute2]);
+
+      router.unregisterRoutes([getRoute2]);
+      router.unregisterRoute(putRoute2);
+
+      expect(router._routes.get('GET')).to.have.members([getRoute1]);
+      expect(router._routes.get('PUT')).to.have.members([putRoute1]);
+    });
+  });
+
+  describe(`setting default and catch handlers`, function() {
```

Comment:
Could we seperate out setDefaultHandler() and setCatchHandler and inline the handling requests piece into these relevant function calls.